### ISSUE TITLE
The `file` access modifier is only valid on top-level types

### ISSUE THREAD

**Issue by BillWagner** [dotnet/docs#40867]
### Type of issue

Typo

### Description

In a couple sentences, the description of the `file` access modifier could be used on "members", either nested types or member fields, properties, or methods. Neither is allowed. The `file` access modifier is allowed only on top-level (non-nested) type declarations.

### Page URL

https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/access-modifiers

### Content source URL

https://github.com/dotnet/docs/blob/main/docs/csharp/programming-guide/classes-and-structs/access-modifiers.md

### Document Version Independent Id

76a389f6-b568-3568-6483-6df440f96f0c

### Article author

@BillWagner

### Metadata

* ID: 8121f526-cfe4-091d-d031-2cdc274cd8ee 
* Service: **dotnet-csharp**
* Sub-service: **fundamentals**

**Comment by bigboybamo** [comment]
Hi @BillWagner I'd like to pick this up but i'm not sure i fully understand the context of what needs to be done.

**Comment by BillWagner** [comment]
Hi @bigboybamo 

The distinction we're missing is that the `file` access modifier is not allowed on any nested types:

```csharp
// This is legal:
file class C
{
    // ...
}

public class C2
{
    // This is not allowed. C3 is not a top level type.
    file class C3
    {
        // ...
    }
}
```

Let me know if you need more details.
I've assigned this issue to you.

**Comment by bigboybamo** [comment]
Hi @BillWagner I've checked again.
I didn't see any section where the file example is defined in https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/access-modifiers which is why I said I wasn't sure earlier

**Comment by BillWagner** [comment]
Hi @bigboybamo 

`file` access isn't super common, so I only added it to the language reference here: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/file

**Comment by bigboybamo** [comment]
Hi @BillWagner I opened a PR #41399. I didn't add the example code because it would cause a build warning. Unless there's some other way to illustrate that?

**Comment by BillWagner** [comment]
Fixed in #41399
